src/App.tsx

```diff
@@ -15,36 +16,104 @@
   'Garlic',
 ];
 
+enum SortType {
+  alph = 'alph',
+  length = 'length',
+}
+
+interface SortParams {
+  sortField: string | number,
+  reversed: boolean
+}
+
+const getPreparedGoods = (goods: string[], { sortField, reversed }: SortParams) => {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.alph:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reversed,
+  });
+
+  const resetSortField = () => {
+    setSortField('');
+    setReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          onClick={() => setSortField(SortType.alph)}
+          className={classNames('button is-info', {
+            'is-light': sortField !== SortType.alph,
+          })}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          onClick={() => setSortField(SortType.length)}
+          className={classNames('button is-success', {
+            'is-light': sortField !== SortType.length,
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={() => setReversed(!reversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField === 'alph' || sortField === 'length' || reversed) && (
+          <button
+            type="button"
+            className={classNames('button is-danger', {
+              'is-light': sortField !== 'n',
```

Comment:
The condition `sortField !== 'n'` in the `className` for the Reset button seems incorrect. It should likely check against `SortType` values instead of `'n'`. Consider revising this condition to ensure it reflects the intended logic.